src/server/hset_family.cc

```diff
@@ -1169,10 +1169,10 @@ void HSetFamily::HRandField(CmdArgList args, const CommandContext& cmd_cntx) {
         }
       }
 
-      if (string_map->Empty()) {
-        auto it_mutable = db_slice.FindMutable(db_context, key, OBJ_HASH);
-        it_mutable->post_updater.Run();
```

Comment:
you removed in few places in this pr calls to post_updater.Run()
and added a comment that it will run immediately.
how will it run? I believe that in the current interface we need to run it before we call del, not so good approach but I think that you break something here if you remove this

Reply:
`ItAndUpdater` is returned by `FindMutable` as a temporary and is destructed right after the call. ItAndUpdater object contains `post_updater` that will run during the destruction. It's not explicit. hence the comment that that explains it.

Reply:
I see now.. thats very confusing

Reply:
I actually borrowed it from Shachar's code from another place. I find it acceptable if it has a comment :)